recipes/thrift/all/conanfile.py

```diff
@@ -95,13 +95,10 @@ def _configure_cmake(self):
         for option, value in self.options.items():
             self._cmake.definitions[option.upper()] = value
 
-        self._cmake.definitions["WITH_SHARED_LIB"] = self.options.shared
-
         # Make thrift use correct thread lib (see repo/build/cmake/config.h.in)
         self._cmake.definitions["USE_STD_THREAD"] = self.options.with_stdthreads
         self._cmake.definitions["USE_BOOST_THREAD"] = self.options.with_boostthreads
-        self._cmake.definitions["WITH_SHARED_LIB"] = self.options.shared
-        self._cmake.definitions["WITH_STATIC_LIB"] = not self.options.shared
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
```

Comment:
This is a standard flag which is managed by Conan

Reply:
@ericLemanissier Okay, I see that you have undone this and re-added the deprecated definitions WITH_SHARED_LIB/WITH_STATIC_LIB

But when Thrift is supporting `BUILD_SHARED_LIBS` why should we use the custom definitions?

Reply:
I did it because I saw in https://thrift.apache.org/lib/cpp that this will be the case in a future 1.0.0 release
it looks like their support for BUILD_SHARED_LIBS is buggy, see https://github.com/conan-io/conan-center-index/pull/1700#pullrequestreview-418667660 and https://github.com/microsoft/vcpkg/blob/master/ports/thrift/portfile.cmake#L23-L24